### ISSUE TITLE
Type consistency for `IntensityType`

### ISSUE THREAD

**Issue by pcolaianni** [OpenMS/OpenMS#3379]
Hey, is there a specific reason for `IntensityType` to be a `float` in `Peak1D` and `Peak2D`, but `double` in `ChromatogramPeak`?
I noticed this while writing a templated method that works on both `MSChromatogram` and `MSSpectrum`.
I save the intensity as `double`, but of course the compiler alerts me with a warning:
```
warning: narrowing conversion of ‘out_ys.std::vector<_Tp, _Alloc>::operator[]<double, std::allocator<double> >(i)’ from ‘__gnu_cxx::__alloc_traits<std::allocator<double> >::value_type {aka double}’ to ‘OpenMS::Peak1D::IntensityType {aka float}’ inside { } [-Wnarrowing]
       typename PeakContainerT::PeakType point { out_xs[i], out_ys[i] };
```

**Comment by cbielow** [comment]
not sure why double would be needed here. Float precision should be more than enough.
Maybe @hroest  has insights?

**Comment by hroest** [comment]
this was done before my time and I think we can change it to `float` -- honestly, even RT may not need to be stored with `double` precision, a `float` would be more than sufficient

**Comment by pasqoo** [comment]
Would you still benefit from solving this issue? I could take care of it these days

**Comment by pasqoo** [comment]
@timosachsenberg Any input regarding this issue?

If it's still needed, I noticed that `typedef double IntensityType;` and `typedef double CoordinateType;` are present in many places.
Should all of those become `float`?

**Comment by timosachsenberg** [comment]
hmm good question. In many cases the intensity is internally an integer. Note that the highest single precision integer a float can represent perfectly is 16777216 - which is quite low.
Do you guys think this might be relevant in practice?

**Comment by hroest** [comment]
@timosachsenberg  I dont think we have run into this issue yet, but 1.6 E7 seems low given that Orbitraps can measure millions of ions at once. However, I think the actual maximum value of FLOAT is 3.40282347e+38F  for 16bit float (https://www.keil.com/support/man/docs/armcc/armcc_chr1359125038425.htm) so you are saying that we just make a small error once we go past 1.6 E7 ? I think we anyways cannot measure intensity to the ppm level so I think that should be fine...


>  Do you guys think this might be relevant in practice?

I guess my answer would be no, not relevant in practice
